Code/Framework/AzToolsFramework/AzToolsFramework/Archive/NullArchiveComponent.cpp

```diff
@@ -16,91 +16,62 @@ namespace AzToolsFramework
     {
         return false;
     }
 
-    void NullArchiveComponent::ListFilesInArchive(const AZStd::string& /*archivePath*/, AZStd::vector<AZStd::string>& /*consoleOutput*/, AZ::Uuid /*taskHandle*/, const ArchiveResponseOutputCallback& respCallback)
+    std::future<bool> NullArchiveComponent::AddFileToArchive(
+        const AZStd::string& /*archivePath*/,
+        const AZStd::string& /*fileToAdd*/,
+        const AZStd::string& /*pathInArchive*/)
     {
-        // Always report we failed to extract
-        AZ::TickBus::QueueFunction(respCallback, false, AZStd::string());
-    }
-
-    bool NullArchiveComponent::ListFilesInArchiveBlocking(const AZStd::string& /*archivePath*/, AZStd::vector<AZStd::string>& /*consoleOutput*/)
-    {
-        return false;
-    }
-
-    void NullArchiveComponent::AddFileToArchive(const AZStd::string& /*archivePath*/, const AZStd::string& /*fileToAdd*/, const AZStd::string& /*pathInArchive*/, AZ::Uuid /*taskHandle*/, const ArchiveResponseOutputCallback& respCallback)
-    {
-        // Always report we failed to extract
-        AZ::TickBus::QueueFunction(respCallback, false, AZStd::string());
-    }
-
-    bool NullArchiveComponent::AddFileToArchiveBlocking(const AZStd::string& /*archivePath*/, const AZStd::string& /*fileToAdd*/, const AZStd::string& /*pathInArchive*/)
-    {
-        return false;
-    }
-
-    bool NullArchiveComponent::AddFilesToArchiveBlocking(const AZStd::string& /*archivePath*/, const AZStd::string& /*workingDirectory*/, const AZStd::string& /*listFilePath*/)
-    {
-        return false;
-    }
-
-    void NullArchiveComponent::AddFilesToArchive(const AZStd::string& /*archivePath*/, const AZStd::string& /*workingDirectory*/, const AZStd::string& /*listFilePath*/, AZ::Uuid /*taskHandle*/, const ArchiveResponseOutputCallback& respCallback)
-    {
-        // Always report we failed to extract
-        AZ::TickBus::QueueFunction(respCallback, false, AZStd::string());
-    }
-
-    void NullArchiveComponent::CreateArchive(const AZStd::string& /*archivePath*/, const AZStd::string& /*dirToArchive*/, AZ::Uuid /*taskHandle*/, const ArchiveResponseOutputCallback& respCallback)
-    {
-        // Always report we failed to extract
-        AZ::TickBus::QueueFunction(respCallback, false, AZStd::string());
-    }
-
-    bool NullArchiveComponent::CreateArchiveBlocking(const AZStd::string& /*archivePath*/, const AZStd::string& /*dirToArchive*/)
-    {
-        return false;
+        return DefaultFuture();
     }
 
-    void NullArchiveComponent::CancelTasks(AZ::Uuid /*taskHandle*/)
+    std::future<bool> NullArchiveComponent::AddFilesToArchive(
+        const AZStd::string& /*archivePath*/,
+        const AZStd::string& /*workingDirectory*/,
+        const AZStd::string& /*listFilePath*/)
     {
+        return DefaultFuture();
     }
 
     void NullArchiveComponent::Reflect(AZ::ReflectContext* context)
```

Comment:
Would it be worth adding a version in the serialize context for this? The changes to this component seem significant enough for a version bump to me.

Reply:
Yep, can do.